### PR TITLE
Add RSpec Unit tests

### PR DESCRIPTION
**Pull request by ankitt26**
### - In this PR, I add the unit tests for all the methods in all the classes:

- create `book_spec.rb` for `Book` class.
- create `person_spec.rb` for `Person` class.
- create `student_spec.rb` for `Student` class.
- create `teacher_spec.rb` for `Teacher` class.
- create `classroom_spec.rb` for `Classroom` class.
- create `rental_spec.rb` for `Rental` class.

**Note** - _I am working alone this week and are not paired up with anyone._

image

### CODE REVIEW

**Comment by nahnahdev** [comment]
## Project Status: INVALIDATED :red_circle: 
Hi @ankitt26 , 

While you made a great effort in this project, I can't go ahead and review your code.


## Invalid Code Review Request

- This is because you have submitted a project, or [pair-programming activity](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/7_mobile_menu.md#work-and-submission-mode) and as a result needs to have commits from both you and your partner or partners. 💯
- However, if for some reason you have to do the project alone, then attach a screenshot of the message you received from `student success` stating that you can do the project alone in the pull request description 👍
- Your Commit History
image
 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ankitt26** [comment]
Hello @NahnahAJ, I attached the screenshot of the email, that I received from student success.

**Comment by nahnahdev** [comment]
Great!!! @ankitt26 You can resubmit it for a review :+1:

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    it 'return the name' do
+      expect(@person.correct_name).to eq(@person.name)
+    end
+  end
+
+  describe '#can_use_services?' do
+    it 'check if age is 18 or greater ' do
+      expect(@person.can_use_services?).to eq(true)
+    end
+  end
```
Comment (AbelG101) [inline spec/person_spec.rb:35]: - [x] Please make sure to add a unit test for the negative scenario as well that is when the person is under 18 years old it should return false. This ensures the `can_use_services` method responds appropriately in all conditions.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      @person.add_rental(rental)
+      expect(@person.rentals).to include(rental)
+    end
+  end
+
+  describe '#correct_name' do
+    it 'return the name' do
+      expect(@person.correct_name).to eq(@person.name)
+    end
+  end
```
Comment (AbelG101) [inline spec/person_spec.rb]: - [x] Kindly make sure to add a unit test for checking the `correct_name` method returns the person name capitalized and trimmed to only 10 characters (If they're longer than 10 characters) this should be implemented in the trimmer decorator class. 😉

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @ankitt26 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All classes have been tested ✔
- Properly written Unit tests ✔
- All tests are passing ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (ankitt26) [reply spec/person_spec.rb:35]: Thanks for reviewing, I will add these tests.

File: spec/person_spec.rb (hunk 2 above)
Comment (ankitt26) [reply spec/person_spec.rb]: hii @AbelG101 , I fixed the `test case`, but the method functionality to capitalize and trim the person's name to only 10 characters is not implemented in the Person class. We cannot capitalize or trim the name from this class. The return of the `correct_name` method should be `@name`, and that is correct. I have checked many colleagues' repositories they also did the same.

**Review by mikethreels — APPROVED** [review]
Hi Ankit, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 2 above)
Comment (AbelG101) [reply spec/person_spec.rb]: - Hey @ankitt26, yes you are correct I seem to have completely confused what I was trying to say and I apologize for that. What I meant to say was that the decorator classes like the `TrimmerDecorator` and `CapitalizeDecorator` should be covered in your tests as part of the project requirements shown below 
  image
- Once again sorry for the confusion and I'd highly advise you to test them as it'll increase the test coverage of your codebase, thank you for your attention to this matter!

Comment (ankitt26) [reply spec/person_spec.rb]: Apologies for confusion. I'll test TrimmerDecorator and CapitalizeDecorator per project requirements. Your input is appreciated. Thank you!
